Marlin/src/module/tool_change.cpp

```diff
@@ -446,11 +446,11 @@ void tool_change(const uint8_t tmp_extruder, const float fr_mm_s/*=0.0*/, bool n
               hotend_offset[Y_AXIS][tmp_extruder] - hotend_offset[Y_AXIS][active_extruder]
             };
 
-            #if ENABLED(MESH_BED_LEVELING)
+            #if HAS_MESH
```

Comment:
The UBL equivalent of lines 458 & 459 need to be added.
```
458  planner.apply_leveling(current_position[X_AXIS], current_position[Y_AXIS], z1);
459  planner.apply_leveling(x2, y2, z2);
```

Reply:
Are you sure about that? `planner.apply_leveling` and `unapply_leveling` include code specific to UBL.

Reply:
This code should be obviated by my proposed change to `tool_change` where it disables leveling before the Z raise and then re-enables leveling for the "move back."

But this means we have to fix **all** cases where there's a Z raise followed by a Z lower, especially where this is done by "spoofing" the current Z position. This is because the spoofing approach is incompatible with Z Fade. So this would also include the Z hop done as part of `G10`/`G11`.

The reason for this is easy to understand. Z fade changes the amount of leveling compensation depending on both the starting and ending Z position of a move. If a move is done by "spoofing" the Z position then the amount of fade will be incorrect for both the start and end of the Z move (and the Z will be slightly off for any moves done in-between the Z raise and Z lower).

These Z moves are only complicated by the fact that we need to keep the same Z position (i.e., spoof Z) as part of the raise and lower behavior. There are a couple of approaches that could fix this:

1. Leverage the Z babystepping code and include it whenever there are (a) multiple tools requiring raise/lower, or (b) firmware retraction. Then we could set the Z babystepping distance as needed.
2. Temporarily set the Z fade to 0 (or just disable leveling) temporarily in-between `G10` and `G11`.
3. Continue to use "spoofing" but make sure it's symmetric: If the Z position is spoofed before the raise, then don't spoof it before the lower, but make sure to only spoof it after the lower. This may not work because the amount of Z correction will at the XY where the `G10` is done will differ from the Z correction where the `G11` is done.

I'm going to have users test and see whether any of these approaches works as part of debugging the current open Z hop and tool change issues.